### ISSUE TITLE
New features, improvements and contribution

### ISSUE THREAD

**Issue by SezerYardim** [Seneral/FlagPlayer#9]
Hi. 

I was looking for an open source project to contribute and i found this app. I really liked the motivation behind it and the benefits because i am struggling with youtube's performance too.  I would like to contribute to it. Do you have any plans to add new features? 

#### My Ideas You May Want To Consider

1. When i want to read some comments on youtube, i am generally curious about the thing in my mind and looking for whether somebody has mentioned it or not. **So maybe adding search bar to the comments section. Or some kind of filtering**. Youtube only has newest and best filters.
2. **I think it is not clear how to use it. At home page there is a short guide but it says "you can copy the playlist url or IDs in the search bar". But you can literally search anything like youtube's search bar**. And i think navigating to youtube, searching what you want to play, clicking it and copying the url is not suitable to the motivation behind because you navitaged to youtube anyways and struggled with its performance.
3. **Even though it is more performant than youtube, the video's loading speed is way more slower from it.**
4. **Adding a new feature that provides adding all the playlists when users copy theirs "playlists" url.**

**Comment by NeuroCPP** [comment]
@Seneral looks like you are getting some attention ....good to know ...Hey
brother you may also consider seeing my issue or enhancement request if you
want to.

☺️

**Comment by Seneral** [comment]
Yes #8 might be an idea, or a way to cache a full playlist at once (5 at a time or something), though error handling will be annoying.
Here's my thoughts on your ideas:
1. Would love that, but unfortunately it won't be 100% possible since youtube doesn't provide that functionality. Obvious idea would be to load all comments, which you can already do by setting it to autoscroll (middle mouse) and letting it sit for a while.
That takes a while but it will be no slower than any alternative approach, really, I tried. What you'll find is that eventually it will fail when there are many comments, which means it is not possible to reliably search anyway. Actually I put in some effort to be able to load even more comments than is possible on youtube itself, though that was with the old way of loading comments so it's not true anymore.
So what I'm saying is, not worth the effort, just load all manually (at least the UI won't slow down quickly like on youtube) and search in page if you really need to.
2. I guess it could be updated, though I thought it rather obvious. There are other aspects though that aren't explained well so that's an idea.
3. That's because it needs to still load the full page, after it loads the flagplayer page. It should be cached though so basically start loading immediately. It might be that there is some delay that could be avoided in the way it's loading parsing metadata and load further comments, though none of it should delay it by much
I think the majority of the perceived delay comes from the fact that youtube plays immediately when the page is loaded, while flagplayer still needs to load the youtube page after the UI is shown already, but the total time is no different or faster.
Just perhaps, another factor is the player used, that flagplayer waits a bit longer for more video/audio to be cached before it starts, but I don't dislike that personally.
4. Wdym? Playlists of users after entering their channel name/id? If so, then yes, that is something I absolutely want to add, but never got to. Involved loading the playlist tab and trying to parse the myriad of combinations the youtube page can be laid out by the channel owner.

One other idea if you want to help is a fix, the comments and recommendations are currently not yet working on mobile afaik. This is due to the last fix which I only tested on desktop and took long enough, though it should be rather straightforward to find the way to parse the comments on mobile (hopefully).

Finally, it currently fails in firefox, if you use that, it seems to get video end events before the video ends, causing a premature skipping to the next video/song, which makes it unusable right now.

Do tell me what you want to work on, might be interested in tackling some of them myself soon

**Comment by NeuroCPP** [comment]
@Seneral maybe you misunderstood my saying...i meant @Sezer-dev to see my enhancement req..

**Comment by Seneral** [comment]
Nah don't worry, understood just fine:)
